nntrainer/layers/flatten_layer.cpp

```diff
@@ -23,26 +23,33 @@ namespace nntrainer {
 static constexpr size_t SINGLE_INOUT_IDX = 0;
 
 void FlattenLayer::finalize(InitLayerContext &context) {
-  ReshapeLayer::setProperty({"target_shape=-1"});
+  const TensorDim &in_dim = context.getInputDimensions()[0];
+
+  std::string target_shape =
+    "target_shape=1:1:" + std::to_string(in_dim.getFeatureLen());
+  ReshapeLayer::setProperty({target_shape});
+
   /** @note the output dimension is in invalid state till finalize of
    * reshape_layer is finished */
   ReshapeLayer::finalize(context);
 
-  const TensorDim &in_dim = context.getInputDimensions()[0];
   if (in_dim.channel() == 1 && in_dim.height() == 1) {
     ml_logw("Warning: the flatten layer is redundant");
   }
 }
 
 void FlattenLayer::setProperty(const std::vector<std::string> &values) {
```

Comment:
Why don't we just simply call Reshape::setProperty?

Reply:
@lhs8928 Umm.. I think 
1. if an error is occurred, we will be able to see where the error happened.
```c++
std::string msg = "[FlattenLayer] Unknown Layer Properties count " + /// please note `FlattenLayer`
                      std::to_string(values.size());
```
2. for extensibility. it would make it easy to extend some logic if it has its own codes in the first place.

Please let me know if I am wrong.